pilot/control/payloads/generic.py

```diff
@@ -28,6 +28,48 @@ def __init__(self, args, job, out, err):
         self.__out = out
         self.__err = err
 
+    def get_job(self):
+        """
+        Get the job object.
+
+        :return: job object.
+        """
+        return self.__job
+
+    def get_queuedata(self):
+        """
+        Get queue data.
+
+        :return: queue data.
+        """
+        return self.__args.location.queuedata
+
+    def get_corecount(self):
+        """
+        Get corecount.
+
+        :return: corecount.
+        """
+        corecount = self.__args.location.queuedata['corecount']
+        if not corecount:
+            corecount = 1
+        return corecount
+
+    def get_es_stageout_gap(self):
```

Comment:
Wen, would you please consider Information component `pilot.info` to resolve `queuedata`  settings.

 I have just pushed the merge request with updates, where I added required fields (corecount, pledgedcpu, zip_time_gap, es_stageout_gap)

So, once Paul merges the request, 
instead of your implemented wrapper `get_es_stageout_gap` with hardcoded validation logic,
you can consider `pilot.info` queuedata.


(which also respects overwrites values for testing coming from the pilot config -- currently they just hardocoded in the `PilotConfigProvider`)
https://github.com/PanDAWMS/pilot2/pull/47/files#diff-e53af538e35e0e36c845ea19ad4edc01R50

You can also drop the wrappers of `Executor` like `get_queuedata`, `get_corecount`, `get_es_stageout_gap` and replace them with direct call to pilot.info.infosys or job['infosys'] depending on if you want to consider job overwrites or not (see below).


an example (equivalent of your `get_es_stageout_gap` function)  for the job related workflow is :

``job['infosys'].queuedata.es_stageout_gap``

in general for non Job-related workflow, queuedata from pilot info could be fetched like this:

```
from pilot.info import infosys
_maxinputsize = infosys.queuedata.maxwdir
```
https://github.com/PanDAWMS/pilot2/blob/next/pilot/util/workernode.py#L66

Reply:
I will update this part.